src/API/private-api.js

```diff
@@ -359,6 +359,54 @@ export default class PrivateAPI {
 		return itemDeltas;
 	}
 
+	static async _updateCurrencies(targetUuid, currencies, userId, {
+		skipVaultLogging = false, interactionId = false
+	} = {}) {
+
+		const targetActor = Utilities.getActor(targetUuid);
+
+		const transaction = new Transaction(targetActor);
+
+		const currenciesToUpdate = PileUtilities.getPriceFromString(currencies).currencies
+			.filter(currency => currency.quantity);
+
+		const itemsToUpdate2 = currenciesToUpdate.filter(currency => currency.type === "item")
+			.map(currency => ({ item: currency.data.item, quantity: currency.quantity }));
+
+		const attributesToUpdate = currenciesToUpdate.filter(currency => currency.type === "attribute")
+			.map(currency => ({ path: currency.data.path, quantity: currency.quantity }));
+
+		await transaction.appendItemChanges(itemsToUpdate2, { type: "currency" });

```

Comment:
This will only add to the item-based currencies, not update like the line below. Perhaps we would need a similar parameter as `set` on item based updates.

Reply:
I'll try to integrate that

Reply:
Not really sure about this... please chekc the commit https://github.com/fantasycalendar/FoundryVTT-ItemPiles/pull/505/commits/fb617b44151ab9617d12138d6c880d53260f78f7